runtime/lua/vim/lsp/client.lua

```diff
@@ -451,13 +449,13 @@ function Client.create(config)
     requests = {},
     attached_buffers = {},
     server_capabilities = {},
-    dynamic_capabilities = lsp._dynamic.new(id),
+    dynamic_capabilities = {},
```

Comment:
Looks like the methods on dynamic capabilities are in use:

https://github.com/search?utf8=%E2%9C%93&q=dynamic_capabilities+language%3Alua&type=code

This would be a breaking change.
Could we keep the interface as is while just moving the logic into the client or same module at least?